### PR TITLE
Add dir path and ignore option

### PR DESCRIPTION
**Pull request by suyanlong**
goimports-reviser -dir-path ./  -project-name github.com/incu6us/goimports-reviser  -ignore v2 -format -rm-unused

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ v2/reviser/testdata/*.go
 v2/reviser/testdata/pkg
 dist
 coverage.txt
+.idea
```
Comment (incu6us) [inline .gitignore:5]: remove it. you can put it to global ignore file and leave project ignore file clear as much as possible

File: main.go [hunk 2]
```diff
@@ -137,37 +157,114 @@ func main() {
 		os.Exit(1)
 	}
 
-	var options reviser.Options
-	if shouldRemoveUnusedImports != nil && *shouldRemoveUnusedImports {
-		options = append(options, reviser.OptionRemoveUnusedImports)
+	do := func(filePath string) {
```
Comment (incu6us) [inline main.go:160]: do what? move it from anonymous func and rename it

File: pkg/astutil/testdata/testdata.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build test
```
Comment (incu6us) [inline pkg/astutil/testdata/testdata.go:1]: format for this is `// +build test` and it is already present

File: main.go [hunk 4]
```diff
@@ -137,37 +157,114 @@ func main() {
...
 	}
+}
 
-	if output == "stdout" {
-		fmt.Print(string(formattedOutput))
-	} else if output == "file" {
-		if !hasChange {
-			return
-		}
+func PWD() string {
```
Comment (incu6us) [inline main.go:234]: is it really need to have publicly?

File: main.go [hunk 5]
```diff
@@ -137,37 +157,114 @@ func main() {
...
 	}
+	pi, err := filepath.Abs(ignore)
+	errorCheck(err)
+	p, err := filepath.Rel(PWD(), pi)
+	errorCheck(err)
+	return strings.Contains(path, p)
+}
 
-	formattedOutput, hasChange, err := reviser.Execute(projectName, filePath, localPkgPrefixes, options...)
+func IsDir(p string) bool {
```
Comment (incu6us) [inline main.go:222]: public?

File: main.go [hunk 6]
```diff
@@ -137,37 +157,114 @@ func main() {
...
+	case dirPath != "":
+		load(dirPath, do)
+	case dirPath == "":
+		do(filePath)
 	}
+}
 
-	if shouldFormat != nil && *shouldFormat {
-		options = append(options, reviser.OptionFormat)
+func IsIgnore(path string) bool {
```
Comment (incu6us) [inline main.go:211]: public?

File: main.go [hunk 7]
```diff
@@ -124,11 +144,11 @@ func main() {
 		return
 	}
 
-	if err := validateRequiredParam(filePath); err != nil {
-		fmt.Printf("%s\n\n", err)
-		printUsage()
-		os.Exit(1)
-	}
+	// if err := validateRequiredParam(filePath); err != nil {
```
Comment (incu6us) [inline main.go:147]: if it is not needed, you can remove it

Comment (incu6us) [reply main.go:147]: but I believe it is needed

File: main.go [hunk 8]
```diff
@@ -38,9 +41,26 @@ var (
 	shouldFormat              *bool
 )
 
-var projectName, filePath, localPkgPrefixes, output string
+var ignore string // 忽略指定的目录
```
Comment (incu6us) [inline main.go:44]: why it is placed separated? all params are defined on 46 line

File: main.go [hunk 9]
```diff
@@ -23,6 +25,7 @@ const (
 	localPkgPrefixesArg    = "local"
 	outputArg              = "output"
 	formatArg              = "format"
+	ignoreArg              = "ignore"
```
Comment (incu6us) [inline main.go:28]: ignore what? naming should be clear for understanding

File: Makefile [hunk 10]
```diff
@@ -15,3 +15,6 @@ release-check:
 .PHONY: release-dry-run
 release-dry-run:
 	@goreleaser release --skip-publish --rm-dist
+
+goimports:
+	@goimports-reviser -dir-path ./  -project-name github.com/incu6us/goimports-reviser  -ignore v2 -format -rm-unused
```
Comment (incu6us) [inline Makefile:20]: ignore what?

**Review by incu6us — CHANGES_REQUESTED** [review]

**Comment by suyanlong** [comment]
@incu6us thanks you for review, now,let's again review.

**Comment by incu6us** [comment]
done in version v3.1.0

**Pull request closed without merging**